### PR TITLE
0.4.6

### PR DESCRIPTION
**Pull request by josefaidt**
- removed optional chaining
- use the "svelte way" of setting the style node
- migrate docs to vite

### CODE REVIEW

**Comment by josefaidt** [comment]
vercel's config will need to be updated for that check to pass

File: docs/package.json [hunk 1]
```diff
@@ -1,28 +1,21 @@
...
-  "dependencies": {
-    "sirv-cli": "^1.0.8"
+    "@rollup/plugin-replace": "^2.4.1",
+    "@sveltejs/vite-plugin-svelte": "^1.0.0-next.5",
+    "autoprefixer": "^10.2.5",
+    "postcss": "^8.2.8",
+    "svelte": "^3.35.0",
+    "svelte-highlight": "^0.7.1",
+    "svelte-hmr": "^0.13.0",
+    "svelte-preprocess": "^4.6.9",
```
Comment (dysfunc) [inline docs/package.json:17]: These aren't required any longer, right?

```suggestion
```

**Comment by josefaidt** [comment]
There appears to be an issue going on with SSR in svelte-kit again.

**Comment by josefaidt** [comment]
@dysfunc I think this is good to go I'm no longer getting errors regarding a default export or SSR related issues in svelte-kit

File: docs/index.html [hunk 2]
```diff
@@ -32,9 +30,9 @@
       content="A theming engine for your Svelte apps using CSS Variables, persisted."
     />
     <meta name="twitter:image" content="favicon.png" />
-
-    <script defer src="/build/bundle.js"></script>
+    <title>Svelte App</title>
```
Comment (dysfunc) [inline docs/index.html:33]:
```suggestion
    <title>Svelte Themer - A theming engine for your Svelte apps using CSS Variables, persisted.</title>
```

File: rollup.config.js [hunk 3]
```diff
@@ -8,17 +8,31 @@ const name = pkg.name
...
+        generate: 'ssr',
+      },
     }),
     resolve(),
     sveld(),
   ],
 }
+
+const preprocessConfig = {
+  input: 'support/preprocess.js',
```
Comment (dysfunc) [inline rollup.config.js:31]: I can't find this file?

Comment (josefaidt) [reply rollup.config.js]: I've commented this out for now. I haven't quite solidified this one just yet.

**Review by dysfunc — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**